### ISSUE TITLE
Issue with mapped types and enums

### ISSUE THREAD

**Issue by blocka** [Microsoft/TypeScript#17288]
**TypeScript Version:**  2.4.1 /

**Code**

```ts
enum Things {
	ONE = "one",
	TWO = "two"
}

type ThingMap = { [TKey in Things]: any };

const thingMap: ThingMap = {
	one: null,
	two: null
};

function setThing(thing: Things, something: any) {
	thingMap[thing] = something;
}

```

**Expected behavior:**
Just like I can do `thingMap[Things.ONE] = 'blah'` without any errors, this should not throw an error.
**Actual behavior:**
src/test.ts(14,2): error TS7017: Element implicitly has an 'any' type because type 'ThingMap' has no index signature.

**Comment by Kovensky** [comment]
Seems related to https://github.com/Microsoft/TypeScript/issues/16687

**Comment by blocka** [comment]
I encountered that too but it's not quite the same thing.

**Comment by mhegazy** [comment]
Actually duplicate of https://github.com/Microsoft/TypeScript/issues/16760, and should be fixed by https://github.com/Microsoft/TypeScript/pull/18029
